packages/react/src/components/Geo/MapView/index.tsx

```diff
@@ -1,11 +1,27 @@
-import { Amplify, Auth } from 'aws-amplify';
+import React, { forwardRef, useEffect, useMemo, useState } from 'react';
 import maplibregl from 'maplibre-gl';
 import { AmplifyMapLibreRequest } from 'maplibre-gl-js-amplify';
-import React, { forwardRef, useEffect, useMemo, useState } from 'react';
 import ReactMapGL from 'react-map-gl';
 import type { MapProps, MapRef, TransformRequestFunction } from 'react-map-gl';
+import { Amplify, Auth } from 'aws-amplify';
 
-export type MapViewProps = Omit<MapProps, 'transformRequest'>;
+// utility types for missing AmplifyConfig type, only includes Geo related key/values
+type AmplifyGeoOptions = {
+  maps?: { default: string };
+  region: string;
+};
+type AmplifyGeoConfig = {
+  geo?: {
+    amazon_location_service?: AmplifyGeoOptions;
+    AmazonLocationService?: AmplifyGeoOptions;
+  };
+};
+
+type MapLib = typeof maplibregl;
+
+interface MapViewProps extends Omit<MapProps, 'transformRequest'> {
+  mapLib: MapLib;
```

Comment:
We can remove the `mapLib` prop since it comes from `MapProps`.

Reply:
`MapProps.mapLib` is typed as `any`, so needed to provide type information to prevent linting issues. Added some more context [here](https://github.com/aws-amplify/amplify-ui/pull/2062/commits/9e73d3e11362f9764cf31a4577d3715694f492d6)